Source/Core/DolphinQt/Config/LogWidget.h

```diff
@@ -43,12 +44,99 @@ class LogWidget final : public QDockWidget, LogListener
   QCheckBox* m_log_wrap;
   QComboBox* m_log_font;
   QPushButton* m_log_clear;
-  QVBoxLayout* m_main_layout;
-  QTextEdit* m_log_text;
-  QWidget* m_tab_log;
+  QPlainTextEdit* m_log_text;
 
   QTimer* m_timer;
 
+  // Ring buffer implementation, heavily based on:
+  // https://embeddedartistry.com/blog/2017/4/6/circular-buffers-in-cc#c-=
+  // Modified to be an almost complete drop-in replacement for std::queue
+  // It does not implement some methods compared to std::queue,
+  // and also implements capacity(), clear() and full(), which std::queue does not
+  template <class T>
+  class circular_buffer
```

Comment:
Our naming conventions for classes is `CamelCase`.

Reply:
I hear you, but I consciously decided against it here because I wanted it to mimick STL containers style as closely as possible.

Reply:
The commit introducing this should also really have rationale or an explanation for its introduction. It may be obvious to you, but not necessarily for others; especially without a justifying reason for it.

Reply:
I pushed a fix for incomplete destructor and also included a rationale in the commit message which should be exhaustive enough. I have **not** renamed the class for reasons given above.

Reply:
`circular_buffer` is now replaced in favour of `FixedSizeQueue`, see below.